### PR TITLE
Add grid-area weighting for EOF / PC

### PR DESCRIPTION
**Pull request by willirath**
Partially solves #80

### CODE REVIEW

**Comment by willirath** [comment]
Needs through checking. But this is as far as I'll get today. @MarcoLandtHayen can you have a look?

File: climate_index_collection/indices.py [hunk 1]
```diff
@@ -1,10 +1,12 @@
 from enum import Enum
 from functools import partial
+from operator import ge
```
Comment (willirath) [inline climate_index_collection/indices.py:3]: ? Should not be here?
```suggestion
```

File: climate_index_collection/grid_handling.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+        Scaled with xy weights.
+
+    """
+    weights = get_xy_weights(
+        lat=dobj.coords["lat"],
+        lon=dobj.coords["lon"],
+        north_lat=north_lat,
+        south_lat=south_lat,
+    )
+    # return weights * dobj
```
Comment (MarcoLandtHayen) [inline climate_index_collection/grid_handling.py:102]: Here you actually want to return ```weights * dogs```, so the hashtag needs to be removed. And the following line can be erased.

**Review by MarcoLandtHayen — COMMENTED** [review]
I really like your approach! But as you write in the code: The weights we get from ```get_xy_weights``` are not normalized. In addition to that - as far as I know - we need to multiply data with ```sqrt``` of the weights before applying SVD. In your code there is no ```sqrt``` and I'm not even sure, where to add it. We find latitude and longitude step size in both models to be almost equidistant. Only for FOCI the latitude steps differ a tiny bit. I propose to assume having equidistant latitude and longitude steps. Then the whole area-weighting can be done in *one line of code* by simply using ```np.sqrt(np.cos(lat))``` as weights. The resulting pc-based indices almost equal the indices obtained from your approach.

**Comment by willirath** [comment]
Let's close here in favor of #85.

**Pull request closed without merging**